### ISSUE TITLE
Admin Search for plugins broken after 9.4.0 Update.

### ISSUE THREAD

**Issue by sverleis** [woothemes/woocommerce#53729]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When upgrading from WooCommerce 9.3.4 to 9.4.0 onwards, the backend search function looks to be broken for the plugins page:

![Image](https://github.com/user-attachments/assets/2ef4443c-729b-4d89-987c-ba8a93af59aa)

Noting console error after the update on load: 

`Store "core/interface" is already registered.` 

And a `500 (Internal Server Error)` when performing a search:

![Image](https://github.com/user-attachments/assets/7ad0f116-b014-4ede-b1f5-dd6446ab4d82)

### Expected behavior

Search to work.

### Actual behavior

Infinite loading after upgrading to 9.4.0 onwards:

![dfUtZF.gif](https://github.com/user-attachments/assets/6510b150-5be3-48fc-979d-477a960f7884)

### Steps to reproduce

1. Set up a site with WooCommerce 9.3.4. Plugin search works.
2. Upgrade to 9.4.0 onwards. Search function breaks.

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://giraffe-of-butterflys.jurassic.ninja
Site address (URL): https://giraffe-of-butterflys.jurassic.ninja
WC Version: 9.4.3
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.2
Log Directory Writable: ✔
WP Version: 6.7.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.3.14
PHP Post Max Size: 2 GB
PHP Time Limit: 850
PHP Max Input Vars: 6144
cURL Version: 8.10.1
OpenSSL/3.0.15

SUHOSIN Installed: –
MySQL Version: 10.11.9-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 9.4.3
WC Database Prefix: wp_
Total Database Size: 2.86MB
Database Data Size: 1.17MB
Database Index Size: 1.69MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.09MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.05MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (4) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.5
Jetpack: by Automattic – 14.2-a.1
WooCommerce: by Automattic – 9.4.3
WP Rollback: by WP Rollback – 2.0.7

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

load.php: by  –

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 14 KB

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/ -  Contains the woocommerce/cart block
Checkout: #7 - /checkout/ -  Contains the woocommerce/checkout block
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.3
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/error.php
/srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/success.php

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: product-data-views
experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
blueprint
reactify-classic-payments-settings
use-wp-horizon

Daily Cron: ✔ Next scheduled: 2024-12-16 05:59:19 +00:00
Options: ✔
Notes: 58
Onboarding: -

### Action Scheduler ###

Complete: 5
Oldest: 2024-12-15 05:59:25 +0000
Newest: 2024-12-15 06:07:57 +0000

Pending: 1
Oldest: 2024-12-16 05:59:25 +0000
Newest: 2024-12-16 05:59:25 +0000

### Status report information ###

Generated at: 2024-12-15 06:21:55 +00:00
`

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by sverleis** [comment]
First reported in ticket: 9162409-zen

**Comment by sverleis** [comment]
Fatal Error log:

```
2024-12-15T06:00:24+00:00 Critical Uncaught Error: Call to a member function get_column_count() on null in /srv/htdocs/wp-content/plugins/woocommerce/src/Packages.php:260  
Additional context
{
    "error": {
        "type": 1,
        "file": "/srv/htdocs/wp-content/plugins/woocommerce/src/Packages.php",
        "line": 260
    },
    "remote-logging": true,
    "backtrace": [
        "",
        "#0 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(326): Automattic\WooCommerce\Packages::display_notice_for_merged_plugins('woocommerce/woo...')",
        "#1 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)",
        "#2 /srv/htdocs/__wp__/wp-includes/plugin.php(517): WP_Hook->do_action(Array)",
        "#3 /srv/htdocs/__wp__/wp-admin/includes/class-wp-plugins-list-table.php(1491): do_action('after_plugin_ro...', 'woocommerce/woo...', Array, 'search')",
        "#4 /srv/htdocs/__wp__/wp-admin/includes/class-wp-plugins-list-table.php(711): WP_Plugins_List_Table->single_row(Array)",
        "#5 /srv/htdocs/__wp__/wp-admin/includes/class-wp-list-table.php(1712): WP_Plugins_List_Table->display_rows()",
        "#6 /srv/htdocs/__wp__/wp-admin/includes/class-wp-list-table.php(1639): WP_List_Table->display_rows_or_placeholder()",
        "#7 /srv/htdocs/__wp__/wp-admin/includes/ajax-actions.php(4834): WP_List_Table->display()",
        "#8 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(324): wp_ajax_search_plugins('')",
        "#9 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)",
        "#10 /srv/htdocs/__wp__/wp-includes/plugin.php(517): WP_Hook->do_action(Array)",
        "#11 /srv/htdocs/__wp__/wp-admin/admin-ajax.php(192): do_action('wp_ajax_search-...')",
        "#12 {main}",
        "thrown"
    ]
}
```

**Comment by tammullen** [comment]
Duplicate of https://github.com/woocommerce/woocommerce/issues/52931
Fixed in 9.5.1
